### DISCUSSION TITLE
The 1.7 RELEASE of Kade Engine

### DISCUSSION THREAD

**Discussion by Kade-github** [KadeArchive/Kade-Engine#1902] [General]
[Changelog](https://kadedev.github.io/Kade-Engine/changelogs/changelog-1.7)

Linux and Mac builds will be distributed at a later date (maybe not mac but linux defn)

<hr /><em>This discussion was created from the release <a href='https://github.com/KadeDev/Kade-Engine/releases/tag/1.7'>The 1.7 RELEASE of Kade Engine</a>.</em>

**Comment by Spel0** [comment]
Epic

**Reply by Sonicjustin95** [reply]
I really am not comfortable with the new update, it crashes a lot and has wayyyyyyyyy too much lag.

**Reply by Spel0** [reply]
Skill issue

**Reply by isakube** [reply]
Skill issue

**Reply by Sonicjustin95** [reply]
really? because other weird stuff happends, like bfs balls coming off.

**Reply by Riskerman21** [reply]
Skill issue

**Reply by isakube** [reply]
> really? because other weird stuff happends, like bfs balls coming off.

Skill issue

**Reply by GuraGurazy** [reply]
Skill issue

**Reply by THEkatinamicrowave** [reply]
Skeel eechue

**Reply by UngofaGamer** [reply]
> I really am not comfortable with the new update, it crashes a lot and has wayyyyyyyyy too much lag.

skill issue

**Reply by NotIlulus** [reply]
i smell skill issue

**Reply by newintownreal** [reply]
skill issue

**Reply by WillTheGrunt** [reply]
skypoi isoieu

**Reply by sebitamontana** [reply]
si da error y mas necesito el 1.5.1 y no esta

**Comment by napstaa967** [comment]
you forgot to add the health bar colors to the changelogs i think

**Comment by haris2010g** [comment]
Ο tord μου αρεσει οπως ειναι απλα μονο βιντεακια χρειαζετε πυστευω

**Reply by Osmaix** [reply]
WTF

**Comment by BombasticTom** [comment]
yooooooooooooooooooooooooooooooo

**Reply by Rogerpro45** [reply]
:v

**Comment by UngofaGamer** [comment]
haha now remove the week locks😐

**Reply by ghost** [reply]
no

**Reply by napstaa967** [reply]
no

**Reply by LukasRainbow** [reply]
no

**Reply by newintownreal** [reply]
no

**Reply by SuxxorState** [reply]
no

**Reply by HAJackReal** [reply]
just remove #if debug and #end in storymenustate.hx

**Reply by isakube** [reply]
no

**Reply by TheSiblingIsBest** [reply]
no

**Reply by NotSorena** [reply]
yes

**Comment by MatteoNobler** [comment]
source/LuaClass.hx:1669: characters 42-54 : Type not found : StatePointer
source/Replay.hx:55: lines 55-145 : Defined in this class
source/LuaClass.hx:1469: characters 42-54 : Type not found : StatePointer
source/Replay.hx:55: lines 55-145 : Defined in this class
source/LuaClass.hx:1144: characters 42-54 : Type not found : StatePointer
source/Replay.hx:55: lines 55-145 : Defined in this class
source/LuaClass.hx:855: characters 41-53 : Type not found : StatePointer
source/Replay.hx:55: lines 55-145 : Defined in this class
source/LuaClass.hx:635: characters 40-52 : Type not found : StatePointer
source/Replay.hx:55: lines 55-145 : Defined in this class
source/LuaClass.hx:370: characters 42-54 : Type not found : StatePointer

How can i fix it?

**Reply by UngofaGamer** [reply]
U need the MVS(Microsoft's visual studio) Setted up and type in the kade folder cmd it shoul pop up the Command promt then run "lime test windows" if that is continuing idrk

**Reply by UngofaGamer** [reply]
Or just install all the libaries

**Reply by MatteoNobler** [reply]
thx,I'll try to find the missing libraries

**Reply by ConnerToons** [reply]
> thx,I'll try to find the missing libraries

Did you ever figure out what was wrong? I'm having the same issue after reinstalling MVS and reinstalling all of the libraries for it

**Reply by MatteoNobler** [reply]
sorry man,I'm still trying to understand the issue, I still have no idea which the problem may be

**Reply by ConnerToons** [reply]
Aw ok, if I find out anything I'll let you know

**Reply by ConnerToons** [reply]
someone figured it out here
https://github.com/KadeDev/Kade-Engine/discussions/1902#discussioncomment-1224529

**Reply by MatteoNobler** [reply]
Thx bro

**Reply by MatteoNobler** [reply]
I've tried,but it continues to throw the same error :(
Even if it didn't work,I apreciated the help, thank you

**Reply by ConnerToons** [reply]
Aw dang, you did the them in the cmd prompt too right?

**Reply by MatteoNobler** [reply]
yep

**Reply by MatteoNobler** [reply]
Don't ask me how,but I repeated the process and now it is compiling XD

**Reply by ConnerToons** [reply]
Yoo epic

**Reply by UngofaGamer** [reply]
> thx,I'll try to find the missing libraries

Actually the haxeflixel libaries are installed

**Comment by saicronise** [comment]
![dsadasda](https://user-images.githubusercontent.com/85856537/130449626-fb817379-5ac8-4989-9ec7-8917646d41f6.png)

This happens when I change the bpm of the song.

The squares no longer keep track of where the note can be placed.

**Reply by LukasRainbow** [reply]
i have the same Problem

**Reply by SuxxorState** [reply]
this actually happens in most cases outside of the KE as well. this is just to keep the notes in the same part of the song, as the BPM affects the chart speed and whatnot (not talking about scroll speed). if it helps i recommend establishing a bpm before charting a song.

**Reply by saicronise** [reply]
> this actually happens in most cases outside of the KE as well. this is just to keep the notes in the same part of the song, as the BPM affects the chart speed and whatnot (not talking about scroll speed). if it helps i recommend establishing a bpm before charting a song.

Actually this is the first time this happened to me. Only after the 1.6 Upadte.

And still happen with the new fix of 1.7

**Comment by tposejank** [comment]
This is epic

**Comment by JackTrials** [comment]
i have no clue on how to change the speed of the song and it is stuck on 1x

**Reply by napstaa967** [reply]
hold shift then press left/right

**Reply by JackTrials** [reply]
:0 tsym!

**Reply by TheCitizen09** [reply]
I can change the Speed and other things but it doesn't save when i press enter, what do i do?
edit: nvm just do it in the freeplay menu instead

**Comment by Goostiny** [comment]
im trying to code in a new character right? i wanna make them an icon so i made an icon png for them since the icons aren't in just 1 png anymore, so i went to HealthIcon.xml and it doesn't have any of those icon settings anymore. how do i add icons to characters now? (i might sound dumb because im not experienced with coding)

**Reply by Spel0** [reply]
Put png in icons at preload/images and name it like this "icon-characterNameInCode". If you want 2 characters to share same icon then you name character like this in code "characterName-charVariant" and leave the same name for icon

**Reply by Goostiny** [reply]
so for example like this? icon-catbobcatbob-mic (don't mind name)

**Reply by Goostiny** [reply]
ok nevermind i got it. thank you!

**Comment by Goostiny** [comment]
source/LuaClass.hx:35: characters 46-58 : Type not found : StatePointer
source/Replay.hx:55: lines 55-145 : Defined in this class
source/LuaClass.hx:1669: characters 42-54 : Type not found : StatePointer
source/Replay.hx:55: lines 55-145 : Defined in this class
source/LuaClass.hx:1469: characters 42-54 : Type not found : StatePointer
source/Replay.hx:55: lines 55-145 : Defined in this class
source/LuaClass.hx:1144: characters 42-54 : Type not found : StatePointer
source/Replay.hx:55: lines 55-145 : Defined in this class
source/LuaClass.hx:855: characters 41-53 : Type not found : StatePointer
source/Replay.hx:55: lines 55-145 : Defined in this class
source/LuaClass.hx:635: characters 40-52 : Type not found : StatePointer
source/Replay.hx:55: lines 55-145 : Defined in this class
source/LuaClass.hx:370: characters 42-54 : Type not found : StatePointer

i got this when i compiled the game. how do i fix it

**Reply by napstaa967** [reply]
haxelib remove linc_luajit
haxelib remove hxvm-luajit
haxelib git linc_luajit https://github.com/nebulazorua/linc_luajit.git
haxelib git hxvm-luajit https://github.com/nebulazorua/hxvm-luajit

**Reply by reyume** [reply]
> haxelib remove linc_luajit
> haxelib remove hxvm-luajit
> haxelib git linc_luajit https://github.com/nebulazorua/linc_luajit.git
> haxelib git hxvm-luajit https://github.com/nebulazorua/hxvm-luajit

lifesaver

**Reply by Unkn0wnGamin** [reply]
it doesn't work for me

**Comment by LukasRainbow** [comment]
So i have a bug to report: When Changing the BPM of the Song the Square you "select" (you hover your mose over it) no longer matches up with the Squares from the Debug Menu. (see pic). It´s hard mapping a song :/

![a](https://user-images.githubusercontent.com/89423600/130523451-610d5b82-b914-455d-8de8-d8ff6fce29db.png)

I really have no idea how to fix it

ALSO Please make an Setting in the Options to turn of Anti Spam. As far as i can tell there is no way to turn it off making some song impossible to complete that we´re possible before :/

P.S. is there an easier way to make a brand new song with brand new characters?
(like for example add a custom bob (from litterally every mod ever) with the ded background? It´s too complicated for my Brain and i can´t even start doing the XML File cuz how tf does it work. h e l p)

P.S. Number 2: can there be a way to make custom colous in the health bar for each character?

P.S. Numero 3: Can you put back "Copy Last Section"? I need it so bad

P.S Numerbro 4: Can you make "Play Hitsounds" so that when start playing in the middle of a song (in the Debug Menu) the sound is instantly there because as of right now the Sounds still fades in for like 5s....

P.S. Nuemafn 5: When you Pause the Song at a specific point (i haven´t found out when) you can complete the Song without doing anything. You don´t get a rating but still... Unpausing the Song causes you all the Notes you would need to hit to Miss and therefore lose the Song if there were too many.

P.S. Idon´tKnowWhyI´mStillDoingThis 6: Fix the Debug menu pls it is very broken

**Comment by BoloVEVO** [comment]
It is me or the arrows appear at very low frame rate while the game is in 120+ fps in fullscreen

**Reply by Yesufz** [reply]
that happens to me too I don't know what causes that

**Comment by ghost** [comment]
Amazing! Only one thing I don't like is the scroll speed. Fix it, there's a blank on the held arrows.

**Comment by Saifix3** [comment]
aye anyone got discord?
i do not see the source file anywhere everytime i download the new releases sometimes not even the game 
![image](https://user-images.githubusercontent.com/89474655/130654774-8e801227-527c-4e7c-b93c-8f5d67f3af4c.png)
ive got everything i got haxedevlop microsoft visuals i used the command prompt thing but its not working im just tryna add a character this is really stress full

**Reply by TheCitizen09** [reply]
what's your discord? I'll help

**Comment by JP1619** [comment]
bruh when i click in the .exe (the game) it just get a black screen then it closes (im using windows 10 pro btw)

**Reply by JP1619** [reply]
i also aleardy tried installing again the kade engine but it dont work

**Reply by rigatoni46** [reply]
Try removing some useless files from your computer.

**Comment by SuxxorState** [comment]
is it just me or do the modcharts not work on the 1.7 version? i feel like im missing something cause they just flat out wont work for me.

**Comment by Switchy143** [comment]
Ok, so I'm trying to implement a new character but it always shows up as daddy dearest when I try to put it in game, here's the code plz help. (I had to put the xml file in word so it would fit into easily manageable pages)
![killme](https://user-images.githubusercontent.com/88582298/131240476-dae30130-b7d5-414b-9885-5fd591ca48c0.PNG)
![killme2 pt1](https://user-images.githubusercontent.com/88582298/131244533-edad27fa-8dcb-4df4-a8ef-b826219ba3a0.PNG)
![killme2 pt2](https://user-images.githubusercontent.com/88582298/131244537-22e8bb29-f520-4702-a33d-3f1fe74741d5.PNG)
![killme2 pt3](https://user-images.githubusercontent.com/88582298/131244542-becb958a-5926-43e1-9e53-908cac56abf6.PNG)
![killme2 pt4](https://user-images.githubusercontent.com/88582298/131244547-f31f0d02-bff7-4f90-a437-2da99c56b2f8.PNG)
![killme2 pt5](https://user-images.githubusercontent.com/88582298/131244551-80de3b0f-1648-45aa-99fb-432c911962fb.PNG)
![killme3](https://user-images.githubusercontent.com/88582298/131244631-2ae6d3d4-c960-408c-ab03-e46179650143.PNG)

**Reply by HugeAngelNT** [reply]
DDDUUUUDDDEEE i have the same problem!!

**Reply by AlansGames** [reply]
does the chart have the correct name

**Reply by AlansGames** [reply]
for the character

**Comment by Switchy143** [comment]


**Reply by Jaduplex** [reply]
1: you dont put the xml in the game code itself lmao
2: you cant add extra lines to offsets.txt that arent offsets

**Reply by Switchy143** [reply]
Yes, I know.

**Comment by HugeAngelNT** [comment]
you know what? 1.7 is trash anyway, im gonna use 1.6.2 instead!

**Reply by rigatoni46** [reply]
N O .

**Reply by HugeAngelNT** [reply]
oh wait nevermind i found a 1.7 tutorial i guess 1.7 isnt trash after all :/

**Reply by UngofaGamer** [reply]
> oh wait nevermind i found a 1.7 tutorial i guess 1.7 isnt trash after all :/

hahahahah xD

**Comment by rigatoni46** [comment]
How do I remove the week locks?

**Reply by UngofaGamer** [reply]
An esay way use botplay

**Reply by UngofaGamer** [reply]
that helped me

**Reply by HugeAngelNT** [reply]
OH ok that explains a lot!!

**Reply by HugeAngelNT** [reply]
im not using botplay and i have no locks at all, it must be my lucky day!

**Reply by UngofaGamer** [reply]
> im not using botplay and i have no locks at all, it must be my lucky day!

wdym?

**Comment by pepperly-art** [comment]
I'm not smart enough to figure out the stage code stuff aaaa
I had some special animations going on in the bg and I'm not sure how the code talks to itself anymore.

![image](https://user-images.githubusercontent.com/82689031/131896298-a3f2a8ea-bbe8-4c01-b9b6-6d57afacb791.png)

**Comment by UngofaGamer** [comment]
Soo i have an error that says that my character needs an offset file, i use whitty but i cant find the offset file in the folders

**Reply by Sazzria** [reply]
i think you need to refer to the xml file and write it yourself (you can copy and paste the other character's offset file but change up the numbers)

**Reply by UngofaGamer** [reply]
> i think you need to refer to the xml file and write it yourself (you can copy and paste the other character's offset file but change up the numbers)

wait nvm i'll try with ayana that is the same offsets as mom

**Comment by UngofaGamer** [comment]
Can Anyone pass me whitty's offsets pls?

**Comment by HugeAngelNT** [comment]
1.how do i change the colors of the characters?

2.how do i add songs?

3.how do i add stages

4.how do i modify the week menu like the agoti mod?

i dunno how to code :(

**Reply by UngofaGamer** [reply]
For the answers the 1, 2, 3 check bright vids they help
And the 4 is the mic'd up engine

**Reply by HugeAngelNT** [reply]
I swear to god, ive searched everywere and found almost absolutly NOTHING!!!!

**Reply by nikodem124** [reply]
vfvg

**Reply by BrandonHatesAnime** [reply]
1. Be more specific, all I can say is if you want to change the physical appearance of a character just mod the sprite sheets or if you want to make a custom character, make your own XML and PNG.
2. Go to assets/songs, make new folders with your song names in it and add the MP3 and OGG inst and voices you want in them
3. Go to source/Stage.hx to add new stages. Hopefully you have the pictures for it. You should base the code off of the dad stage (The one that starts with 'default')
4. Like the guy who first replied to you, that requires Mic'd Up Engine.

If my answers are not your cup of tea, instead of cussing at me and hating this reply go watch BrightFyre's tutorials.  https://www.youtube.com/c/BrightFyre

**Comment by UngofaGamer** [comment]
how do i code the character icon?

**Reply by THEkatinamicrowave** [reply]
Add your character, make the icon thingie like the others, and rename it "[character_name]-icon". then it'll appear when you select the character

**Reply by UngofaGamer** [reply]
ok thx :)

**Comment by Unkn0wnGamin** [comment]
Error: LuaCharacter.cpp
./src/LuaCharacter.cpp(121): error C2039: 'error': is not a member of 'linc::lual'
C:/HaxeToolkit/haxe/lib/linc_luajit/git//linc/linc_lua.h(34): note: see declaration of 'linc::lual'
./src/LuaCharacter.cpp(129): error C3861: 'error': identifier not found
./src/LuaCharacter.cpp(142): error C2039: 'pushcfunction': is not a member of 'linc::lua'
C:/HaxeToolkit/haxe/lib/linc_luajit/git//linc/linc_lua.h(21): note: see declaration of 'linc::lua'
./src/LuaCharacter.cpp(145): error C3861: 'pushcfunction': identifier not found
./src/LuaCharacter.cpp(150): error C2039: 'error': is not a member of 'linc::lual'
C:/HaxeToolkit/haxe/lib/linc_luajit/git//linc/linc_lua.h(34): note: see declaration of 'linc::lual'
./src/LuaCharacter.cpp(153): error C3861: 'error': identifier not found
./src/LuaCharacter.cpp(166): error C2039: 'pushcfunction': is not a member of 'linc::lua'
C:/HaxeToolkit/haxe/lib/linc_luajit/git//linc/linc_lua.h(21): note: see declaration of 'linc::lua'
./src/LuaCharacter.cpp(169): error C3861: 'pushcfunction': identifier not found
./src/LuaCharacter.cpp(174): error C2039: 'error': is not a member of 'linc::lual'
C:/HaxeToolkit/haxe/lib/linc_luajit/git//linc/linc_lua.h(34): note: see declaration of 'linc::lual'
./src/LuaCharacter.cpp(177): error C3861: 'error': identifier not found
./src/LuaCharacter.cpp(182): error C2039: 'pushcfunction': is not a member of 'linc::lua'
C:/HaxeToolkit/haxe/lib/linc_luajit/git//linc/linc_lua.h(21): note: see declaration of 'linc::lua'
./src/LuaCharacter.cpp(185): error C3861: 'pushcfunction': identifier not found
./src/LuaCharacter.cpp(190): error C2039: 'error': is not a member of 'linc::lual'
C:/HaxeToolkit/haxe/lib/linc_luajit/git//linc/linc_lua.h(34): note: see declaration of 'linc::lual'
./src/LuaCharacter.cpp(193): error C3861: 'error': identifier not found
./src/LuaCharacter.cpp(198): error C2039: 'pushcfunction': is not a member of 'linc::lua'
C:/HaxeToolkit/haxe/lib/linc_luajit/git//linc/linc_lua.h(21): note: see declaration of 'linc::lua'
./src/LuaCharacter.cpp(201): error C3861: 'pushcfunction': identifier not found
./src/LuaCharacter.cpp(206): error C2039: 'error': is not a member of 'linc::lual'
C:/HaxeToolkit/haxe/lib/linc_luajit/git//linc/linc_lua.h(34): note: see declaration of 'linc::lual'
./src/LuaCharacter.cpp(209): error C3861: 'error': identifier not found
./src/LuaCharacter.cpp(214): error C2039: 'pushcfunction': is not a member of 'linc::lua'
C:/HaxeToolkit/haxe/lib/linc_luajit/git//linc/linc_lua.h(21): note: see declaration of 'linc::lua'
./src/LuaCharacter.cpp(217): error C3861: 'pushcfunction': identifier not found
./src/LuaCharacter.cpp(222): error C2039: 'error': is not a member of 'linc::lual'
C:/HaxeToolkit/haxe/lib/linc_luajit/git//linc/linc_lua.h(34): note: see declaration of 'linc::lual'
./src/LuaCharacter.cpp(225): error C3861: 'error': identifier not found
./src/LuaCharacter.cpp(308): error C2039: 'error': is not a member of 'linc::lual'
C:/HaxeToolkit/haxe/lib/linc_luajit/git//linc/linc_lua.h(34): note: see declaration of 'linc::lual'
./src/LuaCharacter.cpp(308): error C3861: 'error': identifier not found
./src/LuaCharacter.cpp(351): error C2039: 'error': is not a member of 'linc::lual'
C:/HaxeToolkit/haxe/lib/linc_luajit/git//linc/linc_lua.h(34): note: see declaration of 'linc::lual'
./src/LuaCharacter.cpp(351): error C3861: 'error': identifier not found
./src/LuaCharacter.cpp(380): error C2039: 'error': is not a member of 'linc::lual'
C:/HaxeToolkit/haxe/lib/linc_luajit/git//linc/linc_lua.h(34): note: see declaration of 'linc::lual'
./src/LuaCharacter.cpp(380): error C3861: 'error': identifier not found
./src/LuaCharacter.cpp(408): error C2039: 'error': is not a member of 'linc::lual'
C:/HaxeToolkit/haxe/lib/linc_luajit/git//linc/linc_lua.h(34): note: see declaration of 'linc::lual'
./src/LuaCharacter.cpp(408): error C3861: 'error': identifier not found
./src/LuaCharacter.cpp(439): error C2039: 'error': is not a member of 'linc::lual'
C:/HaxeToolkit/haxe/lib/linc_luajit/git//linc/linc_lua.h(34): note: see declaration of 'linc::lual'
./src/LuaCharacter.cpp(439): error C3861: 'error': identifier not found
./src/LuaCharacter.cpp(468): error C2039: 'error': is not a member of 'linc::lual'
C:/HaxeToolkit/haxe/lib/linc_luajit/git//linc/linc_lua.h(34): note: see declaration of 'linc::lual'
./src/LuaCharacter.cpp(468): error C3861: 'error': identifier not found

 - src/flixel/input/gamepad/mappings/FlxGamepadMapping.cpp
 - src/haxe/ds/_List/ListNode.cpp

How to fix?

**Comment by Brad2024** [comment]
I was looking through the files, and I noticed that the icons for the tankman are there. Is KadeDev planning to add week 7 to Kade Engine? If so, then will he also add the week 7 firework arrows? I want the firework arrows.

**Reply by isakube** [reply]
lmao no they were there before

**Comment by Zulfigithud** [comment]
u should add note splashes

**Reply by UngofaGamer** [reply]
There is a way, grab the note splashes from an engine or mod, put it in shared (also put the xml) and done!

**Comment by UngofaGamer** [comment]
![image](https://user-images.githubusercontent.com/88444471/136707299-9f046955-da09-4a21-8d83-27762f84419a.png)
help me pls
